src/Expression/Type.hs

```diff
@@ -0,0 +1,17 @@
+module Expression.Type where
+
+import Builder
```

Comment:
Again, I think `Builder.Type` is sufficient.

Reply:
Ah, sorry, ignore the above comment, apparently there is no `Builder.Type`. Or does it make sense to add it as well, for consistency?

Reply:
Looking at `Stage` below, I think we should only add `.Type` modules if they help resolve a cyclic import. So, let's not add `Builder.Type` if it's not necessary.

Reply:
Indeed I didn't add `.Type` modules that were not in @angerman's PR under `src/Types/`, and the ones that were there presumably were there to resolve cyclic imports, so I think the PR effectively implements what you converged on in your last comment, doesn't it?